src/Http/Middleware/AddCustomProvider.php

```diff
@@ -51,4 +56,26 @@ public function terminate()
     {
         config(['auth.guards.api.provider' => $this->defaultApiProvider]);
     }
+
+    /**
+     * Check if the given provider is not registered in the auth configuration file.
+     *
+     * @param $provider
+     *
+     * @return bool
+     */
+    protected function invalidProvider($provider)
+    {
+        if (is_null($provider)) {
+            return true;
+        }
+
+        foreach (config('auth.guards') as $guardsConfiguration) {
+            if ($guardsConfiguration['provider'] === $provider) {
+                return false;
+            }
+        }
+
+        return false;
```

Comment:
@Alymosul Has an error here. If you pass a provider that not found on `config('auth.guards')` the `OAuthServerException` should be thrown. But returning false the provider don't will be invalid.